tests/integration/cartography/intel/github/test_repos.py

```diff
@@ -390,6 +394,79 @@ def test_python_library_in_multiple_requirements_files(neo4j_session):
     assert node_ids == {"okta", "okta|0.9.0"}
 
 
+@patch.object(cartography.intel.github.repos, "transform")
+@patch.object(cartography.intel.github.repos, "get")
+@patch.object(cartography.intel.github.repos, "_get_repo_collaborators")
+def test_collabs_sync(
+    mock_repo_collaborators, mock_get_repos, mock_transfrom, neo4j_session
+):
+    """
+    Ensure that sync sends the right collaborator data to transform
+    """
+
+    # load hjsimpson lmsimpson mbsimpson
+    test_users._ensure_local_neo4j_has_test_data(neo4j_session)
+
+    # Return an empty result from transform
+    mock_transfrom.return_value = {
+        "repos": [],
+        "repo_languages": [],
+        "repo_owners": [],
+        "repo_outside_collaborators": {},
+        "repo_direct_collaborators": {},
+        "python_requirements": [],
+        "dependencies": [],
+        "manifests": [],
+    }
+
+    # return the normal repos
+    mock_get_repos.return_value = GET_REPOS
+
+    # mock the collaborators - numbers don't match the GET_REPOS data, but should be ok for a quick test.
+    # As long as there is  contributor in the repo data, we will fetch them
+    repo_users = {
+        GET_REPOS[1]["name"]: ["hjsimpson"],
+        GET_REPOS[2]["name"]: ["lmsimpson"],
+    }
+
+    def collaborators_side_effect(*args, **kwargs):
+        repo = args[3]
+        affiliation = args[4]
+        if affiliation == "OUTSIDE":
+            return PaginatedGraphqlData(nodes=[], edges=[])
+        users = repo_users.get(repo, [])
+        edges = [{"permission": "ADMIN"} for _user in users]
+        nodes = [
+            {
+                "url": f"https://github.com/{user}",
+                "login": user,
+                "name": None,
+                "email": "",
+                "company": None,
+            }
+            for user in users
+        ]
+        return PaginatedGraphqlData(nodes=nodes, edges=edges)
+
+    mock_repo_collaborators.side_effect = collaborators_side_effect
+
+    cartography.intel.github.repos.sync(
+        neo4j_session, TEST_JOB_PARAMS, "some_key", "http://localhost", "simpsoncorp"
+    )
+
+    # transform takes 3 args
```

Comment:
Instead of testing the call_args, can we test outcomes instead? 

Specifically, this test should let us reproduce the bug you observed. We should be able to take the cartography master branch, and apply just this test without any other changes, run it, and then see that contributors are attached to things they should not be attached to.